x-pack/plugins/fleet/public/applications/fleet/sections/settings/components/edit_output_flyout/advanced_options_section.tsx

```diff
@@ -0,0 +1,303 @@
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFormRow>
+        <EuiSpacer size="m" />
+        <EuiFormRow
+          fullWidth
+          {...memQueueEvents.formRowProps}
+          label={
+            <FormattedMessage
+              id="xpack.fleet.settings.editOutputFlyout.memQueueEventsLabel"
+              defaultMessage="Memory Queue Events"
+            />
+          }
+        >
+          <EuiFlexGroup alignItems="flexStart">
+            <EuiFlexItem>
+              <EuiFieldNumber {...memQueueEvents.props} placeholder="Events" min={0} />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.fleet.settings.editOutputFlyout.memQueueEventsSizeDescription"
+                  defaultMessage="Maximum number of events that can be stored in the queue. Default is set to 4096."
+                />
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFormRow>
+
+        <EuiHorizontalRule />
+
+        <EuiFormRow fullWidth {...diskQueueEnabledInput.formRowProps}>
+          <EuiFlexGroup alignItems="flexStart">
+            <EuiFlexItem>
+              <EuiSwitch
+                data-test-subj="editOutputFlyout.diskQueueSwitch"
+                {...diskQueueEnabledInput.props}
+                label={
+                  <FormattedMessage
+                    id="xpack.fleet.settings.editOutputFlyout.diskQueueSwitchLabel"
+                    defaultMessage="Disk Queue"
+                  />
+                }
+              />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.fleet.settings.editOutputFlyout.diskQueueSwitchDescription"
+                  defaultMessage="Once enabled, events will be queued on disk when there's a requirement to cache them on the agents in the event of connection loss."
+                />
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFormRow>
+        <EuiSpacer size="m" />
+
+        <EuiFormRow fullWidth {...diskQueueEncryptionEnabled.formRowProps}>
+          <EuiFlexGroup alignItems="flexStart">
+            <EuiFlexItem>
+              <EuiSwitch
+                data-test-subj="editOutputFlyout.diskQueueEncryption"
+                {...diskQueueEncryptionEnabled.props}
+                label={
+                  <FormattedMessage
+                    id="xpack.fleet.settings.editOutputFlyout.diskQueueEncryptionLabel"
+                    defaultMessage="Encryption"
+                  />
+                }
+              />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.fleet.settings.editOutputFlyout.diskQueueEncryptionDescription"
+                  defaultMessage="Enable encryption of data at rest."
```

Comment:
I think this should be more specific about what data we're talking about.

```suggestion
                  defaultMessage="Enable encryption of data written to the disk queue."
```